### DISCUSSION TITLE
did this Package work with mongodb

### DISCUSSION THREAD

**Discussion by idrisalshikh** [romanzipp/Laravel-Queue-Monitor#93] [Q&A]
when I try with mongodb I get exceptions
Call to a member function prepare() on null
so is this package should work with nun relation database like MongoDB out of the box or not

**Comment by romanzipp** [comment]
I'm assuming you're using jenssegers/mongodb?
Please post a full stacktrace of the exception and provide some steps to reproduce this issue.

**Reply by idrisalshikh** [reply]
PHP 8.1.2
composer create-project laravel/laravel:^8.x TestQueueMonitorWithMongodb8
composer require jenssegers/mongodb
composer require romanzipp/laravel-queue-monitor
I configured the database and the queue
then I created exampleJobs this job class execute insert statements to the MongoDB
I called this job from my route /
then I start the queue job 
everything works fine and I can see the inserted data in the database
I added the IsMonitored to my job class and visit my main route and it work again with no problem 
when I added the jobs route
Route::prefix('jobs')->group(function () {
    Route::queueMonitor();
});
and visit this route I get this exception

`Call to a member function prepare() on null {"exception":"[object] (Error(code: 0): Call to a member function prepare() on null at E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Database\\Connection.php:368)
[stacktrace]
#0 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Database\\Connection.php(705): Illuminate\\Database\\Connection->Illuminate\\Database\\{closure}('select count(*)...', Array)
#1 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Database\\Connection.php(672): Illuminate\\Database\\Connection->runQueryCallback('select count(*)...', Array, Object(Closure))
#2 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Database\\Connection.php(376): Illuminate\\Database\\Connection->run('select count(*)...', Array, Object(Closure))
#3 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Database\\Query\\Builder.php(2385): Illuminate\\Database\\Connection->select('select count(*)...', Array, true)
#4 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Database\\Query\\Builder.php(2373): Illuminate\\Database\\Query\\Builder->runSelect()
#5 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Database\\Query\\Builder.php(2907): Illuminate\\Database\\Query\\Builder->Illuminate\\Database\\Query\\{closure}()
#6 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Database\\Query\\Builder.php(2374): Illuminate\\Database\\Query\\Builder->onceWithColumns(Array, Object(Closure))
#7 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Database\\Query\\Builder.php(2521): Illuminate\\Database\\Query\\Builder->get()
#8 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Database\\Query\\Builder.php(2480): Illuminate\\Database\\Query\\Builder->runPaginationCountQuery(Array)
#9 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Database\\Eloquent\\Builder.php(808): Illuminate\\Database\\Query\\Builder->getCountForPagination()
#10 E:\\TestQueueMonitorWithMongodb8\\vendor\
omanzipp\\laravel-queue-monitor\\src\\Controllers\\ShowQueueMonitorController.php(56): Illuminate\\Database\\Eloquent\\Builder->paginate(35)
#11 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\ControllerDispatcher.php(48): romanzipp\\QueueMonitor\\Controllers\\ShowQueueMonitorController->__invoke(Object(Illuminate\\Http\\Request))
#12 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Route.php(262): Illuminate\\Routing\\ControllerDispatcher->dispatch(Object(Illuminate\\Routing\\Route), Object(romanzipp\\QueueMonitor\\Controllers\\ShowQueueMonitorController), '__invoke')
#13 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Route.php(205): Illuminate\\Routing\\Route->runController()
#14 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Router.php(721): Illuminate\\Routing\\Route->run()
#15 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(128): Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#16 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Middleware\\SubstituteBindings.php(50): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#17 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(167): Illuminate\\Routing\\Middleware\\SubstituteBindings->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#18 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken.php(78): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#19 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#20 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\View\\Middleware\\ShareErrorsFromSession.php(49): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#21 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(167): Illuminate\\View\\Middleware\\ShareErrorsFromSession->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#22 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Session\\Middleware\\StartSession.php(121): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#23 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Session\\Middleware\\StartSession.php(64): Illuminate\\Session\\Middleware\\StartSession->handleStatefulRequest(Object(Illuminate\\Http\\Request), Object(Illuminate\\Session\\Store), Object(Closure))
#24 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(167): Illuminate\\Session\\Middleware\\StartSession->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#25 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse.php(37): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#26 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(167): Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#27 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Cookie\\Middleware\\EncryptCookies.php(67): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#28 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(167): Illuminate\\Cookie\\Middleware\\EncryptCookies->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#29 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#30 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Router.php(723): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#31 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Router.php(698): Illuminate\\Routing\\Router->runRouteWithinStack(Object(Illuminate\\Routing\\Route), Object(Illuminate\\Http\\Request))
#32 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Router.php(662): Illuminate\\Routing\\Router->runRoute(Object(Illuminate\\Http\\Request), Object(Illuminate\\Routing\\Route))
#33 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Router.php(651): Illuminate\\Routing\\Router->dispatchToRoute(Object(Illuminate\\Http\\Request))
#34 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Kernel.php(167): Illuminate\\Routing\\Router->dispatch(Object(Illuminate\\Http\\Request))
#35 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(128): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}(Object(Illuminate\\Http\\Request))
#36 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#37 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsToNull.php(31): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#38 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsToNull->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#39 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#40 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Middleware\\TrimStrings.php(40): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#41 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\TrimStrings->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#42 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#43 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#44 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance.php(86): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#45 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#46 E:\\TestQueueMonitorWithMongodb8\\vendor\\fruitcake\\laravel-cors\\src\\HandleCors.php(38): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#47 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(167): Fruitcake\\Cors\\HandleCors->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#48 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Http\\Middleware\\TrustProxies.php(39): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#49 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(167): Illuminate\\Http\\Middleware\\TrustProxies->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#50 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#51 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Kernel.php(142): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#52 E:\\TestQueueMonitorWithMongodb8\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Kernel.php(111): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter(Object(Illuminate\\Http\\Request))
#53 E:\\TestQueueMonitorWithMongodb8\\public\\index.php(52): Illuminate\\Foundation\\Http\\Kernel->handle(Object(Illuminate\\Http\\Request))
#54 E:\\TestQueueMonitorWithMongodb8\\server.php(21): require_once('E:\\\\TestQueueMon...')
#55 {main}
"} `

I can't open the monitor page
